### PR TITLE
Final sanity check of project before presentation

### LINKED ISSUE #85: Test integration between all the different micro-services
**Description**
For now most of the services have been developed in isolation, 

**Tasks**
- [ ] Test for all use cases

### PR DESCRIPTION
**Pull request by xandervedder**
## Changes
Lots of small changes that will go towards "completing" the project

## Checklist
- [x] Fixed integration between Customer and Review
- [x] Fixed integration between Order, Customer and Review
- [x] Check if there are more integrations that need to be tested
- [x] Remove delivery if order was deleted
- [x] Remove unused events in Customer

## When merging, this pull request 
closes #85

### CODE REVIEW

File: delivery/src/main/java/nl/softwarestrijders/waiter/delivery/infrastructure/driver/messaging/RabbitMqEventListener.java [hunk 1]
```diff
@@ -15,8 +16,11 @@ public RabbitMqEventListener(DeliveryCommandHandler commandHandler) {
     }
 
     @RabbitListener(queues = "#{'${messaging.queue.order}'}")
-    void listen(DeliveryStatusEvent event) {
-        if ("order.created".equals(event.eventKey)) {
+    void listen(Message message, DeliveryStatusEvent event) {
+        var key = message.getMessageProperties().getReceivedRoutingKey();
+
+        // TODO: delivery should be deleted when an order is deleted, no?
```
Comment (JustMilan) [inline delivery/src/main/java/nl/softwarestrijders/waiter/delivery/infrastructure/driver/messaging/RabbitMqEventListener.java:22]: Shouldn't this be resolved before merge?

**Review by JustMilan — CHANGES_REQUESTED** [review]

File: delivery/src/main/java/nl/softwarestrijders/waiter/delivery/infrastructure/driver/messaging/RabbitMqEventListener.java (hunk 1 above)
Comment (xandervedder) [reply delivery/src/main/java/nl/softwarestrijders/waiter/delivery/infrastructure/driver/messaging/RabbitMqEventListener.java:22]: Ideally, yes, but this is still a draft. Might implement this tommorow or not

Comment (xandervedder) [reply delivery/src/main/java/nl/softwarestrijders/waiter/delivery/infrastructure/driver/messaging/RabbitMqEventListener.java]: This is implemented now

**Review by Rubenvdbrink — APPROVED** [review]
lgtm, sonarcloud detected some things in delivery tho, not sure if u already noticed.

**Comment by xandervedder** [comment]
> lgtm, sonarcloud detected some things in delivery tho, not sure if u already noticed.

I did, might fix this

dont merge yet tho

**Comment by xandervedder** [comment]
> lgtm, sonarcloud detected some things in delivery tho, not sure if u already noticed.

This is fixed now

**Review by JustMilan — APPROVED** [review]

**Review by arjennorbart — APPROVED** [review]

**Comment by xandervedder** [comment]
Waiting for @Rubenvdbrink

File: customer/postman/customer-micro-service.postman_collection.json [hunk 2]
```diff
@@ -227,4 +225,4 @@
 			]
 		}
 	]
-}
+}
```
Comment (Rubenvdbrink) [inline customer/postman/customer-micro-service.postman_collection.json:228]: newline

File: customer/src/main/java/nl/softwarestrijders/waiter/customer/core/domain/Customer.java [hunk 3]
```diff
@@ -143,17 +143,24 @@ public void addOrders(List<UUID> orderIds) {
...
-		this.reviews.remove(reviewId);
+	public void removeReview(Review review) {
+		this.reviews.remove(review);
+	}
+
+	public Review findReviewById(UUID id) {
+		return this.getReviews().stream()
+				.filter(review -> review.reviewId().equals(id))
+				.findFirst()
+				.orElseThrow(() -> new RuntimeException("No review found"));
```
Comment (Rubenvdbrink) [inline customer/src/main/java/nl/softwarestrijders/waiter/customer/core/domain/Customer.java:163]: custom exception?

File: review/src/main/java/nl/softwarestrijders/waiter/review/core/application/CommandHandler.java [hunk 4]
```diff
@@ -6,33 +6,60 @@
...
     public Review handle(CreateReview command) {
         var customer = command.customerId();
         var concept = command.conceptId();
-        if (this.repository.existsByConceptIdAndCustomerId(customer, customer)) {
+        if (this.reviewRepository.existsByConceptIdAndCustomerId(customer, customer)) {
             throw new AlreadyReviewedException(concept, customer);
         }
 
+        if (!this.customerRepository.existsById(customer)) {
+            throw new IllegalArgumentException(String.format("Customer with id %s does not exist", customer));
```
Comment (Rubenvdbrink) [inline review/src/main/java/nl/softwarestrijders/waiter/review/core/application/CommandHandler.java:56]: custom exception?

**Review by Rubenvdbrink — APPROVED** [review]
3 small things but good enough

**Comment by Rubenvdbrink** [comment]
just merge it @xandervedder

**Pull request merged**